### PR TITLE
Output canonical address string (40 hex chars) instead of 7 in logs

### PR DESCRIPTION
**Pull request by avive**
## Motivation
Output canonical address string (last 40 bytes of public key, hex formatted) instead of 7 for easier extraction of user addresses from log files and address related debugging. See #2180.

## Changes
Changed log related formatting calls to use a.String() instead of a.Short(). 

## TODO
- [x] Explain motivation or link existing issue(s)
- [ ] Test changes and document test plan
- [x] Update documentation as needed

### CODE REVIEW

File: api/grpc_server.go [hunk 1]
```diff
@@ -174,15 +174,15 @@ func (s SpacemeshGrpcService) SubmitTransaction(ctx context.Context, in *pb.Sign
 		return nil, err
 	}
 	log.Info("GRPC SubmitTransaction to address: %s (len: %v), amount: %v gaslimit: %v, fee: %v",
-		tx.Recipient.Short(), len(tx.Recipient), tx.Amount, tx.GasLimit, tx.Fee)
+		tx.Recipient.String(), len(tx.Recipient), tx.Amount, tx.GasLimit, tx.Fee)
```
Comment (lrettig) [inline api/grpc_server.go]: You can leave out the changes to this file since it'll go away soon in #2131 (and this would cause a merge conflict)

Comment (avive) [reply api/grpc_server.go]: done

**Comment by noamnelke** [comment]
I don't have a strong objection to the change itself, other than increasing the log size.

However, I strongly disagree with this as a reasonable solution to what's described in the motivation. It should be almost trivial to write a gRPC method that returns a csv- or json-formatted list of balances (full address + balance). It should be just as easy to create a similar list of rewards received, if needed (e.g. coinbase account + total rewards).

Unless you have another reason to merge this change, I think we should implement what I suggested.

**Comment by avive** [comment]
I think we should do both. Log size issue should be dealt on with setting proper log levels on various nodes - not by making log data obscure...

**Comment by lrettig** [comment]
Agree. While I don't think we should print e.g. the full node id everywhere in the logs, in the case of account address, there is a standard format here and I think that should be used in the logs.

**Review by noamnelke — APPROVED** [review]
Fair enough, I concede.

**Comment by avive** [comment]
bors merge

**Comment by avive** [comment]
Too many changes to develop including rename of grpc server file name. It is easier to re-do this change from current develop then to try to merge this oldish pr.

**Pull request closed without merging**